File: plugins/clock/clock.1s.go
//go:build ignore

package main

import (
	"fmt"
	"os/exec"
	"strings"
	"time"
)

func crossbar(args string) string {
	cmd := exec.Command("crossbar", strings.Split(args, " ")...)
	out, err := cmd.Output()
	if err != nil {
		return ""
	}
	return strings.TrimSpace(string(out))
}

func main() {
	now := time.Now()
	
	timeStr := crossbar("--time")
	if timeStr == "" {
		timeStr = now.Format("15:04:05")
	}
	
	dateStr := crossbar("--time --format date")
	if dateStr == "" {
		dateStr = now.Format("2006-01-02")
	}
	
	tz := crossbar("--timezone")
	if tz == "" {
		tz = now.Location().String()
	}

	fmt.Printf("🕐 %s\n", timeStr)
	fmt.Println("---")
	fmt.Printf("Time: %s\n", timeStr)
	fmt.Printf("Date: %s\n", dateStr)
	fmt.Printf("Timezone: %s\n", tz)
	fmt.Println("---")
	fmt.Println("Refresh | refresh=true")
}

File: plugins/clock/clock.1s.rs
//! Clock Plugin - Shows current time using Crossbar API

use std::process::Command;
use chrono::Local;

fn crossbar(args: &str) -> Option<String> {
    let output = Command::new("crossbar")
        .args(args.split_whitespace())
        .output()
        .ok()?;
    
    if output.status.success() {
        Some(String::from_utf8_lossy(&output.stdout).trim().to_string())
    } else {
        None
    }
}

fn main() {
    let now = Local::now();
    
    let time_str = crossbar("--time")
        .unwrap_or_else(|| now.format("%H:%M:%S").to_string());
    let date_str = crossbar("--time --format date")
        .unwrap_or_else(|| now.format("%Y-%m-%d").to_string());
    let tz = crossbar("--timezone")
        .unwrap_or_else(|| now.format("%Z").to_string());

    println!("🕐 {}", time_str);
    println!("---");
    println!("Time: {}", time_str);
    println!("Date: {}", date_str);
    println!("Timezone: {}", tz);
    println!("---");
    println!("Refresh | refresh=true");
}
